src/util/check.h

```diff
@@ -80,4 +80,21 @@ T&& inline_assertion_check(T&& val, [[maybe_unused]] const char* file, [[maybe_u
  */
 #define Assume(val) inline_assertion_check<false>(val, __FILE__, __LINE__, __func__, #val)
 
+/** Helper for CheckNonFatal() */
+template <typename T>
+T&& inline_check_non_fatal(T&& val, [[maybe_unused]] const char* file, [[maybe_unused]] int line, [[maybe_unused]] const char* func, [[maybe_unused]] const char* assertion)
+{
+    CHECK_NONFATAL(val);
+    return std::forward<T>(val);
+}
+
+/**
+ * CheckNonFatal is the identity function. Throws a NonFatalCheckError if value compares equal to zero
+ *
+ * - Should be used to run non-fatal checks.
+ * - For fatal errors, use Assert().
+ * - Appropriate for non-fatal errors in interactive sessions (e.g. RPC or command line interfaces).
+ */
+#define CheckNonFatal(val) inline_check_non_fatal(val, __FILE__, __LINE__, __func__, #val)
```

Comment:
I think you can just recycle the existing name. No need to add a second one.

Reply:
It's now replaced
I had to add unreachable return statements like below otherwise the CI would complain
https://github.com/bitcoin/bitcoin/blob/0661baffe8044cc049f6494ab29b79f5e4b00998/src/rpc/util.cpp#L986-L993